Importamos bibliotecas y datos a utilizar

In [1]:
import pandas as pd

In [2]:
airplanes_dataset=pd.read_csv('./accidentesoriginal.csv')
add=pd.read_csv('./este.csv')

Revisamos duplicados

In [4]:
def revisaduplicados(df,col):
    total=df[df.duplicated(col)]
    
    return total

In [5]:
revisaduplicados(airplanes_dataset,'Unnamed: 0')

Empty DataFrame
Columns: [Unnamed: 0, fecha, HORA declarada, Ruta, OperadOR, flight_no, route, ac_type, registration, cn_ln, all_aboard, PASAJEROS A BORDO, crew_aboard, cantidad de fallecidos, passenger_fatalities, crew_fatalities, ground, summary]
Index: []

# Data Exploring

Con pandas.DataFrame.info pueden ver cuántas filas, columnas, valores no nulos y uso de memoria hay en total.

In [6]:
airplanes_dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5008 entries, 0 to 5007
Data columns (total 18 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Unnamed: 0              5008 non-null   int64 
 1   fecha                   5008 non-null   object
 2   HORA declarada          5008 non-null   object
 3   Ruta                    5008 non-null   object
 4   OperadOR                5008 non-null   object
 5   flight_no               5008 non-null   object
 6   route                   5008 non-null   object
 7   ac_type                 5008 non-null   object
 8   registration            5008 non-null   object
 9   cn_ln                   5008 non-null   object
 10  all_aboard              5008 non-null   object
 11  PASAJEROS A BORDO       5008 non-null   object
 12  crew_aboard             5008 non-null   object
 13  cantidad de fallecidos  5008 non-null   object
 14  passenger_fatalities    5008 non-null   object
 15  crew

Podemos ver que el dataset contiene 5008 filas y 18 columnas.
Para visualizar los datos faltantes usamos isnull().sum()




In [7]:
airplanes_dataset.isnull().sum()

Unnamed: 0                0
fecha                     0
HORA declarada            0
Ruta                      0
OperadOR                  0
flight_no                 0
route                     0
ac_type                   0
registration              0
cn_ln                     0
all_aboard                0
PASAJEROS A BORDO         0
crew_aboard               0
cantidad de fallecidos    0
passenger_fatalities      0
crew_fatalities           0
ground                    0
summary                   0
dtype: int64

Notamos que los datos nulos o faltantes estan con "?"

In [8]:
airplanes_dataset['HORA declarada'].value_counts()

?       1504
1500      35
1400      30
1100      30
1700      29
        ... 
2129       1
1002       1
1151       1
0909       1
1914       1
Name: HORA declarada, Length: 1217, dtype: int64

# Data Cleaning

Lo primero a realizar es seleccionar las columnas a utilizar. Como la finalidad de este trabajo es la realizacion de un DashBoard, elimine las que me parecieron menos redundantes

In [9]:
data = airplanes_dataset.drop(columns=['Unnamed: 0','HORA declarada', 'registration', 'cn_ln', 'registration', 'cn_ln', 'summary','PASAJEROS A BORDO','crew_aboard','passenger_fatalities','crew_fatalities','flight_no','route'])

In [10]:
data.fecha.value_counts()

August 31, 1988       4
September 11, 2001    4
February 23, 1945     3
December 28, 1946     3
August 27, 1992       3
                     ..
December 04, 1958     1
December 23, 1958     1
December 24, 1958     1
December 26, 1958     1
July 06, 2021         1
Name: fecha, Length: 4577, dtype: int64

Empezamos a remplazar los "?" de nuestras columnas

In [13]:
def Replace(dataframe,col,to_rep,replaced_by):
     dataframe[col]=dataframe[col].astype('string').str.replace(to_rep,replaced_by)



In [14]:
Replace(data,'fecha','?','',)
Replace(data,'Ruta','?','',)
Replace(data,'ac_type','?','',)
Replace(data,'OperadOR','?','',)
Replace(data,'cantidad de fallecidos','?','',)
Replace(data,'ground','?','',)
Replace(data,'ac_type','?','',)



C:\Users\Agustina\AppData\Local\Temp\ipykernel_13540\3208315954.py:2: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  dataframe[col]=dataframe[col].astype('string').str.replace(to_rep,replaced_by)


Cambiamos el formato de fecha

In [15]:
data['fecha']=data['fecha'] = pd.to_datetime(airplanes_dataset.fecha, errors = 'coerce')

In [16]:
data.to_csv('./final1.csv', sep=',',index=False)

In [17]:
archivo1=pd.read_csv('./final1.csv')


# Cargamos las tablas con Sqlalchemy

In [18]:
from sqlalchemy import create_engine


In [19]:
conexion_sql= 'mysql+pymysql://root:123456789@localhost:3306/airplanes'

conectar= create_engine(conexion_sql)

def CargaTablasMySql(entry,nombre):  
    
    
    entry.to_sql(name=nombre, con=conectar)




In [20]:
CargaTablasMySql(archivo1,'Tabla1')

C:\Users\Agustina\AppData\Roaming\Python\Python310\site-packages\pandas\io\sql.py:1663: UserWarning: The provided table name 'Tabla1' is not found exactly as such in the database after writing the table, possibly due to case sensitivity issues. Consider using lower case table names.
  warnings.warn(msg, UserWarning)
